Extension/src/Utility/Filesystem/ripgrep.ts

```diff
@@ -0,0 +1,149 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { strict } from 'assert';
+import { existsSync } from 'fs';
+import { accumulator } from '../Async/iterators';
+import { logAndReturn } from '../Async/returns';
+import { Process } from '../Process/process';
+import { ProcessFunction, Program } from '../Process/program';
+import { Instance } from '../System/types';
+import { filepath } from './filepath';
+
+let ripgrep: Instance<ProcessFunction> | undefined;
+export async function initRipGrep(filename: string) {
+    if (!ripgrep) {
+        const rg = await filepath.isExecutable(filename);
+        strict(rg, `File ${filename} is not executable`);
+        ripgrep = await new Program(filename);
+    }
+}
+
+export class FastFinder implements AsyncIterable<string> {
+    private keepOnlyExecutables: boolean;
+    private executableExtensions = new Array<string>();
+    private processes = new Array<Instance<Process>>();
+    private pending = 0;
+    private readyToComplete = false;
+    private distinct = new Set<string>();
+
+    #files = accumulator<string>().autoComplete(false);
+
+    [Symbol.asyncIterator](): AsyncIterator<string> {
+        this.readyToComplete = true;
+        if (this.pending === 0) {
+            this.#files.complete();
+        }
+        // this.#files.complete();

```

Comment:
Is `// this.#files.complete();` intentional? Is it related to the TODO below it or not?

Reply:
I don't think I need that anymore